tests/Integration/RequestIntegrationTest.php

```diff
@@ -40,9 +41,9 @@ public function testInvokeWithRequestHavingIpAddress(bool $shouldSendPii, array
         $event->getUserContext()->setData(['foo' => 'bar']);
 
         $request = new ServerRequest('GET', new Uri('http://www.example.com/fo'), [], null, '1.1', ['REMOTE_ADDR' => '127.0.0.1']);
-        $integration = new RequestIntegration(new Options(['send_default_pii' => $shouldSendPii]));
+        $integration = new RequestIntegration(new Options(['send_default_pii' => $shouldSendPii]), $this->createRequestFactory($request));
 
-        RequestIntegration::applyToEvent($integration, $event, $request);
+        RequestIntegration::applyToEvent($integration, $event);
```

Comment:
Can you please refactor this test method so that we really test the full flow of how the integration would be executed (take a look at the `IgnoreErrorsIntegrationTest` class to see what I mean)?